### ISSUE TITLE
how do I choose which checkpoint to use for translation

### ISSUE THREAD

**Issue by aastha19** [OpenNMT/OpenNMT-py#1637]
hi

I have trained my model for 100000 steps and have checkpoints save every 5000 steps.
how do I choose which model/checkpoint to use for translation?

Is there any way to help decide?

**Comment by francoishernandez** [comment]
If you performed validation during training, you can refer to the reported accuracy and perplexity as a first basis.

Then, you'll need to have some test sets (preferably not in your training data) that you'll run inference on, and score with a metric (BLEU, TER, ROUGE, Levenshtein, ...). There is no perfect automated metric for translation tasks, but it helps validating your checkpoints and hint towards the better one(s).
